components/webgpu/wgpu_thread.rs

```diff
@@ -868,21 +872,70 @@ impl WGPU {
                             );
                         };
                     },
+                    WebGPURequest::BeginRenderPass {
+                        command_encoder_id,
+                        render_pass_id,
+                        label,
+                        color_attachments,
+                        depth_stencil_attachment,
+                        device_id: _device_id,
+                    } => {
+                        let global = &self.global;
+                        let desc = &RenderPassDescriptor {
+                            label,
+                            color_attachments: color_attachments.into(),
+                            depth_stencil_attachment: depth_stencil_attachment.as_ref(),
+                            timestamp_writes: None,
+                            occlusion_query_set: None,
+                        };
+                        let (pass, error) = gfx_select!(
+                            command_encoder_id => global.command_encoder_create_render_pass_dyn(
+                                command_encoder_id,
+                                desc,
+                        ));
+                        assert!(
+                            self.render_passes
+                                .insert(render_pass_id, (pass, error.is_none()))
+                                .is_none(),
+                            "RenderPass should not exist yet."
```

Comment:
But is this checked at `BeginRenderPass`?

Reply:
I am not sure I understand, this is in `BeginRenderPass`.

Reply:
I mean: how can we make an assertion here, unless the corresponding method in `script` disallows this behavior.

Reply:
I just followed here what we done for compute pass. In script the caller (same method) creates new id and then sends ipc message.

Reply:
Ah ok so it would always be a new id.